pkg/core/chain/chain.go

```diff
@@ -392,23 +396,46 @@ func (c *Chain) processCandidateVerificationRequest(r rpcbus.Request) {
 	r.RespChan <- rpcbus.Response{Resp: nil, Err: err}
 }
 
-// Send Inventory message to all peers
-func (c *Chain) advertiseBlock(b block.Block) error {
-	msg := &peermsg.Inv{}
-	msg.AddItem(peermsg.InvTypeBlock, b.Header.Hash)
+// startBlockBroadcast propagates block hash to gossip network and complete block to kadcast network
+func startBlockBroadcast(publisher eventbus.Publisher, b block.Block, kadHeight byte) {
 
-	buf := new(bytes.Buffer)
-	if err := msg.Encode(buf); err != nil {
-		log.Panic(err)
+	// (Re)Propagate complete block data into kadcast network together with kad height
+	if config.Get().Kadcast.Enabled && kadHeight <= kadcast.InitHeight {
+
+		buf := new(bytes.Buffer)
+		if err := message.MarshalBlock(buf, &b); err != nil {
+			log.Panic(err)
+		}
+
+		if err := topics.Prepend(buf, topics.Block); err != nil {
+			log.Panic(err)
+		}
+
+		m := message.NewWithHeader(topics.Block, *buf, []byte{kadHeight})
+		publisher.Publish(topics.Kadcast, m)
+
+		// time.Sleep(1 * time.Second)
```

Comment:
Should we keep this commented line here?

Reply:
That's a good point. I postponed this for now due to a  hesitation:

If allow the delay here this would affect `AcceptBlock` perf (an extra 1 sec added). If we move `startBlockBroadcast` into a separate goroutine, this would change the AcceptBlock behaviour a bit (e.g TestAcceptFromPeer would fail)

What is your opinion on that? Do you see problems with goroutine solution (other than the test failure) or you think extra second delay would not impact AcceptBlock (respectively consensus).

Reply:
Just to clarify: my reply above is about is it safe to uncomment `time.Sleep(1 * time.Second)`

Reply:
I would rather have a goroutine-based solution - but maybe it makes more sense then to do something like `go startBlockBroadcast` and uncommenting the sleep, so that it does not slow down the consensus (it will restart after `AcceptBlock` returns).